lte/gateway/c/oai/tasks/mme_app/mme_app_bearer.c

```diff
@@ -2629,6 +2629,181 @@ void mme_app_handle_nw_init_ded_bearer_actv_req(
+
+  /* If delete_default_bearer is set and this is the only active PDN,
+  *  Send Detach Request to UE
+  */
+  if ((nw_init_bearer_deactv_req_p->delete_default_bearer) &&
+       (ue_context_p->nb_active_pdn_contexts == 1)) {
+    OAILOG_INFO(
+      LOG_MME_APP,
+      "Send detach to NAS for EBI %d as delete_default_bearer = true\n",
+      nw_init_bearer_deactv_req_p->ebi[0]);
+    //Send Deatch Request to NAS
+    if (ue_context_p->ecm_state == ECM_CONNECTED) {
+      mme_app_send_nas_detach_request(
+        ue_context_p->mme_ue_s1ap_id, MME_INITIATED_EPS_DETACH);
+    } else {
+      //If UE is in IDLE state send Paging Req
+      mme_app_paging_request_helper(
+        ue_context_p, true, false /* s-tmsi */, CN_DOMAIN_PS);
+      // Set the flag and send detach to UE after receiving service req
+      ue_context_p->emm_context.nw_init_bearer_deactv = true;
+    }
+  } else {
+    /*If UE is in connected state send Deactivate Bearer Req
+     * + ERAB Rel Cmd to NAS
+     */
+    if (ue_context_p->ecm_state == ECM_CONNECTED) {
+      message_p =
+      itti_alloc_new_message(TASK_MME_APP, MME_APP_DELETE_DEDICATED_BEARER_REQ);
+      if (message_p == NULL) {
+        OAILOG_INFO(
+        LOG_MME_APP,
+        "itti_alloc_new_message failed for"
+        "MME_APP_DELETE_DEDICATED_BEARER_REQ\n");
+        OAILOG_FUNC_OUT(LOG_MME_APP);
+      }
+      MME_APP_DELETE_DEDICATED_BEARER_REQ(message_p).delete_default_bearer =
+        nw_init_bearer_deactv_req_p->delete_default_bearer;
+      MME_APP_DELETE_DEDICATED_BEARER_REQ(message_p).ue_id =
+        ue_context_p->mme_ue_s1ap_id;
+      MME_APP_DELETE_DEDICATED_BEARER_REQ(message_p).no_of_bearers =
+        nw_init_bearer_deactv_req_p->no_of_bearers;
+      memcpy(
+        &MME_APP_DELETE_DEDICATED_BEARER_REQ(message_p).ebi,
+        nw_init_bearer_deactv_req_p->ebi,
+        sizeof(ebi_t));
+      // Send MME_APP_DELETE_DEDICATED_BEARER_REQ to NAS
+      itti_send_msg_to_task(TASK_NAS_MME, INSTANCE_DEFAULT, message_p);
+    } else {
+      /* If UE is in IDLE state remove bearer context
+       * send Create Bearer Rsp to SPGW
+       */
+      for (i = 0; i < nw_init_bearer_deactv_req_p->no_of_bearers; i++) {
+        if ((ue_context_p->bearer_contexts[i]) &&
+          (ue_context_p->bearer_contexts[i]->ebi ==
+            nw_init_bearer_deactv_req_p->ebi[i])) {
+          mme_app_free_bearer_context(&ue_context_p->bearer_contexts[i]);
+          break;
+        }
+      }
+      if (i < nw_init_bearer_deactv_req_p->no_of_bearers) {
+        //Send delete_dedicated_bearer_rsp to SPGW
+        _send_delete_dedicated_bearer_rsp(
+          ue_context_p,
+          nw_init_bearer_deactv_req_p->delete_default_bearer,
+          nw_init_bearer_deactv_req_p->ebi,
+          nw_init_bearer_deactv_req_p->no_of_bearers,
+          pdn_context->s_gw_teid_s11_s4,
+          REQUEST_ACCEPTED);
+      } else {
+        OAILOG_ERROR(LOG_MME_APP, "Cannot delete bearer context");
```

Comment:
More informative log would be nice. Reason, number of bearer_contexts, no_of_bearers

Reply:
done